packages/fastify-renderer/src/node/renderers/react/ReactRenderer.tsx

```diff
@@ -69,14 +58,111 @@ export class ReactRenderer implements Renderer {
       for (const renderable of renderables) {
         await this.loadModule(this.entrypointRequirePathForServer(renderable))
       }
+
+      const mode = this.options.mode
+      this.transformHooks = this.plugin.hooks
+        .map(unthunk)
+        .map((hook) => hook.transform?.absolutePath)
+        .flatMap((hook) => (hook ? [hook] : []))
+
+      const modulePaths = await this.getPreloadPaths()
+      const paths = [...modulePaths, ...this.transformHooks]
+
+      this.workerPool = await createPool({
+        create() {
+          const workerData = {
+            paths,
+          }
+
+          let worker: Worker
+          switch (mode) {
+            case 'streaming':
+              worker = new Worker(require.resolve('./StreamingWorker.import.js'), {
+                workerData,
+              })
+            case 'sync':
+              worker = new Worker(require.resolve('./StaticWorker.import.js'), {
+                workerData,
+              })
+          }
+
+          return worker
+        },
+        async dispose(worker) {
+          await worker.terminate()
```

Comment:
This is current not used. Implemented just in case a need arises for automatic scaling of some kind.